proposals/NNNN-resource-attributes-in-dxil-and-spirv.md

```diff
@@ -0,0 +1,386 @@
+| Texture1D                          | SRV  | vec4   | -   | 1D   | -   | -        | -     | -   | -   |
+| Texture1DArray                     | SRV  | vec4   | -   | 1D   | -   | -        | yes   | -   | -   |
+| Texture2D                          | SRV  | vec4   | -   | 2D   | -   | -        | -     | -   | -   |
+| Texture2DArray                     | SRV  | vec4   | -   | 2D   | -   | -        | yes   | -   | -   |
+| Texture2DMS                        | SRV  | vec4   | -   | 2D   | yes | -        | -     | -   | -   |
+| Texture2DMSArray                   | SRV  | vec4   | -   | 2D   | yes | -        | yes   | -   | -   |
+| Texture3D                          | SRV  | vec4   | -   | 3D   | -   | -        | -     | -   | -   |
+| TextureCUBE                        | SRV  | vec4   | -   | Cube | -   | -        | -     | -   | -   |
+| TextureCUBEArray                   | SRV  | vec4   | -   | Cube | -   | -        | yes   | -   | -   |
+| RWTexture1D                        | UAV  | vec4   | -   | 1D   | -   | -        | -     | -   | -   |
+| RWTexture1DArray                   | UAV  | vec4   | -   | 1D   | -   | -        | yes   | -   | -   |
+| RWTexture2D                        | UAV  | vec4   | -   | 2D   | -   | -        | -     | -   | -   |
+| RWTexture2DArray                   | UAV  | vec4   | -   | 2D   | -   | -        | yes   | -   | -   |
+| RWTexture2DMS                      | UAV  | vec4   | -   | 2D   | yes | -        | -     | -   | -   |
+| RWTexture2DMSArray                 | UAV  | vec4   | -   | 2D   | yes | -        | yes   | -   | -   |
+| RWTexture3D                        | UAV  | vec4   | -   | 3D   | -   | -        | -     | -   | -   |
+| RasterizerOrderedTexture1D         | UAV  | vec4   | yes | 1D   | -   | -        | -     | -   | -   |
+| RasterizerOrderedTexture1DArray    | UAV  | vec4   | yes | 1D   | -   | -        | yes   | -   | -   |
+| RasterizerOrderedTexture2D         | UAV  | vec4   | yes | 2D   | -   | -        | -     | -   | -   |
+| RasterizerOrderedTexture2DArray    | UAV  | vec4   | yes | 2D   | -   | -        | yes   | -   | -   |
+| RasterizerOrderedTexture3D         | UAV  | vec4   | yes | 3D   | -   | -        | -     | -   | -   |
+| FeedbackTexture2D                  | SRV  | -      | -   | -    | -   | fbtype   | -     | -   | -   |
+| FeedbackTexture2DArray             | SRV  | -      | -   | -    | -   | fbtype   | yes   | -   | -   |
+| Buffer                             | SRV  | vec4   | -   | -    | -   | -        | -     | -   | -   |
+| RWBuffer                           | UAV  | vec4   | -   | -    | -   | -        | -     | -   | -   |
+| RasterizerOrderedBuffer            | UAV  | vec4   | yes | -    | -   | -        | -     | -   | -   |
+| ByteAddressBuffer                  | SRV  | -      | -   | -    | -   | -        | -     | yes | -   |
+| RWByteAddressBuffer                | UAV  | -      | -   | -    | -   | -        | -     | yes | -   |
```

Comment:
ByteAddressBuffers are hard in SPIR-V. Without untyped pointers, we cannot cast from one pointer type to another. This make it very hard to do some thing, and impossible to do others. For, example, it is impossible to have both 32-bit atomic operations and 64-bit atomic operations.

For now, we cannot have a common implementation for all of the raw buffer resources.